rzup/rzup-init.sh

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env bash
+# Copyright 2024 RISC Zero, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# Example URLs:
+# for a 64-bit Linux system (x86_64):
+# BASE_URL/rzup-x86_64-linux
+# for a 64-bit macOS system (arm64):
+# BASE_URL/rzup-arm64-darwin
+
+# NOTE: Temporary release URL
+RZUP_BINARY_UPDATE_ROOT="${RZUP_BINARY_UPDATE_ROOT:-https://github.com/hmrtn/asset-test/releases/download/test1/}"
+QUIET=no
+
+# Set XDG_DATA_HOME based on OS
+_ostype="$(uname -s)"
+case "$_ostype" in
+  Darwin)
+    XDG_DATA_HOME="${XDG_DATA_HOME:-${HOME}/Library/Application Support}"
+    ;;
+  *)
+    XDG_DATA_HOME="${XDG_DATA_HOME:-${HOME}/.local/share}"
```

Comment:
Why here and not in home?

Reply:
It can be. But I chose this because it is what the spec calls for to avoid polluting the user home. 

> If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

https://xdgbasedirectoryspecification.com/